panels/sharing/cc-sharing-panel.c

```diff
@@ -646,6 +657,132 @@ cc_sharing_panel_setup_media_sharing_dialog (CcSharingPanel *self)
+{
+  GVariant *discoverable_prop;
+  gboolean content_sharing_enabled;
+  GtkWidget *w;
+
+  discoverable_prop = g_variant_lookup_value (changed_properties, "Discoverable", G_VARIANT_TYPE_BOOLEAN);
+  if (!discoverable_prop)
+    return;
+
+  content_sharing_enabled = g_variant_get_boolean (discoverable_prop);
+  g_variant_unref (discoverable_prop);
+
+  w = GTK_WIDGET (gtk_builder_get_object (self->priv->builder, "content-sharing-switch"));
+  gtk_switch_set_active (GTK_SWITCH (w), content_sharing_enabled);
+}
+
+static gboolean
+cc_sharing_panel_check_content_sharing_available (void)
+{
+  FlatpakInstallation *installation = flatpak_installation_new_system (NULL, NULL);
+  GError              *error = NULL;
+  FlatpakInstalledRef *ref = flatpak_installation_get_installed_ref (installation,
+                                                                     FLATPAK_REF_KIND_APP,
+                                                                     "com.endlessm.CompanionAppService",
+                                                                     NULL,
+                                                                     NULL,
+                                                                     NULL,
+                                                                     &error);
+  gboolean ret = TRUE;
+
+  if (ref == NULL)
+    {
+      if (!g_error_matches (error, FLATPAK_ERROR, FLATPAK_ERROR_NOT_INSTALLED))
+        {
+          g_message ("Unexpected error occurred when checking flatpak installation: %s", error->message);
+        }
+
+      ret = FALSE;
+      g_clear_error (&error);
+    }
+
+  g_object_unref (ref);
+  g_object_unref (installation);
+
+  return ret;
+}
+
+
+static void
+cc_sharing_panel_setup_content_sharing_dialog (CcSharingPanel *self)
+{
+  CcSharingPanelPrivate *priv = self->priv;
+  gboolean content_sharing_enabled;
+  GError *error = NULL;
+  GVariant *value;
+  GtkSwitch *content_sharing_switch = GTK_SWITCH (WID ("content-sharing-switch"));
+
+  self->priv->companion_app_avahi_helper_proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+                                                                                G_DBUS_PROXY_FLAGS_NONE,
+                                                                                NULL,
+                                                                                "com.endlessm.CompanionAppServiceAvahiHelper",
+                                                                                "/com/endlessm/CompanionAppServiceAvahiHelper",
+                                                                                "com.endlessm.CompanionApp.AvahiHelper",
+                                                                                NULL, &error);
+  if (error != NULL)
+    {
+      g_critical ("Unable to create a DBus proxy for the companion app avahi helper: %s", error->message);
+      g_error_free (error);
+
+      content_sharing_enabled = FALSE;
```

Comment:
In this case, you probably also need to make the switch insensitive or something like that -- otherwise we'll crash when trying to access the proxy pointer in the switch callback.
Or even better, fold in this check in the function that determines the visibility of the row; if we fail to create the proxy, it's probably best to hide the whole thing altogether.